src/App.tsx

```diff
@@ -1,15 +1,51 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [chosenPerson, setChosenPerson] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const handleQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!query && !e.target.value.trim()) {
+      return;
+    }
+
+    setQuery(e.target.value);
+    applyQuery(e.target.value);
+    setChosenPerson('');
+  };
```

Comment:
The function name 'handleQueryChange' doesn't follow the naming conventions for event handler functions in React as outlined [here](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364). According to these conventions, the function should be named 'onQueryChange'.